### PR TITLE
[Speedy]  improve upgrade failure error message

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala [hunk 1]
```diff
@@ -213,6 +216,32 @@ private[lf] object Pretty {
...
+                keyOpt,
+                _,
+              ) =>
+            text("Validation fails when trying to upgrade the contract") & prettyContractId(
+              coid
+            ) & text("from") & prettyTypeConName(srcTemplateId) & text("to") & prettyTypeConName(
+              dstTemplateId
+            ) /
+              text(
+                "Verify that neither the signatories, nor the observer, nor the contract key have changds"
```
Comment (basvangijzel-DA) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala:234]: You should also mention maintainers here.

File: daml-lf/transaction/src/main/scala/com/daml/lf/Error.scala [hunk 2]
```diff
@@ -5,8 +5,9 @@ package com.daml.lf
 package interpretation
 
 import com.daml.lf.data.Ref.{ChoiceName, Location, Party, TypeConName}
-import com.daml.lf.transaction.{GlobalKey, NodeId}
+import com.daml.lf.transaction.{NodeId, GlobalKey}
```
Comment (basvangijzel-DA) [inline daml-lf/transaction/src/main/scala/com/daml/lf/Error.scala:8]: Fix import gone wrong?

File: compiler/scenario-service/protos/scenario_service.proto [hunk 3]
```diff
@@ -274,6 +274,14 @@ message ScenarioError {
     Identifier required_interface_id = 3;
   }
 
+  message UpgradeValidationFailed {
+    ContractRef contract_ref = 1;
+    Identifier srcTemplateId =2 ;
+    repeated string signatories = 4;
+    repeated string observers = 5;
+    KeyWithMaintainers key = 6;
```
Comment (basvangijzel-DA) [inline compiler/scenario-service/protos/scenario_service.proto:282]: I think it's worth filing an issue for adding the handling of the ensures clause.

**Review by basvangijzel-DA — APPROVED** [review]
Thanks Rémy!

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala [hunk 4]
```diff
@@ -231,7 +231,7 @@ private[lf] object Pretty {
               dstTemplateId
             ) /
               text(
-                "Verify that neither the signatories, nor the observer, nor the contract key have changds"
+                "Verify that neither the signatories, nor the observer, nor the contract key, not key's mattnainer have changed"
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala:234]:
```suggestion
                "Verify that neither the signatories, nor the observers, nor the contract key, nor the key's maintainers have changed"
```

**Pull request merged**